CBinding/CMake/CMakeProject.cs

```diff
@@ -261,13 +261,26 @@ void AddFile (FilePath fileName, string targetName)
 
 				FileService.CreateDirectory (file.ParentDirectory.Combine (outputDirectory));
 
+				Stream generationResult;
+				int buildFlag = 0;
```

Comment:
The `buildFlag` variable name isn't clear at all. I would have used something like `bool useMSBuild`.

Instead of calling different commands based on the compiler class, I would suggest moving that logic _to the compiler_.

e.g. add the following methods to ICompiler:

```
Task<BuildResult> GenerateMakefiles (string projectName, FilePath outputDirectory, IProgressMonitor monitor);
Task<BuildResult> Build (string projectName, FilePath outputDirectory, IProgressMonitor monitor);
```

Then the MsvcCompiler can implement these to generate MSBuild files and build them, the gnu compiler can implement this to use makefiles, etc.

And then this code can simply do:

```
var results = await compiler.GenerateMakefiles(fileFormat.ProjectName, outputDirectory, monitor);
```

(etc)

I would also suggest moving ExecuteCommand and ParseGenerationResult to be protected methods on `https://github.com/mhutch/cbinding/blob/9a61ad0d6c60fc97a8ab1ccf75d1c6b336fe16d6/CBinding/Compiler/CCompiler.cs`.